client/wsclient.go

```diff
@@ -85,7 +87,41 @@ func (c *wsClient) Stop(ctx context.Context) error {
 	conn := c.conn
 	c.connMutex.RUnlock()
 
+	ticker := time.NewTicker(50 * time.Millisecond)
+
+	// Wait for all remaining messages to be sent. Continuing to attempt
+	// to send messages after calling Stop will eventually result in lost
+	// messages.
+	select {
+	case <-ticker.C:
+		if !c.sender.NextMessage().IsPending() {
```

Comment:
Is this essentially a sleep of 50ms?

Reply:
In most cases it will be, but I have seen this fire multiple times when running this code with a debugger.